File: introductionToAlgorithmCLRS/chp2/Ex_2.1/q2RevInsertionSort.java
import java.util.Scanner;
import java.util.Arrays;

public class q2RevInsertionSort {

    public static int[] sorting (int[] array) {
        int sortedArray[] = array;
        for(int j = 1; j < sortedArray.length; j++) {
            int key = sortedArray[j];
            int i = j - 1 ;
            while (i >-1 && sortedArray[i] < key ) {
                sortedArray[i + 1] = sortedArray[i];
                i = i - 1;
            }
            sortedArray[i+1] = key;
        }
        
        return(sortedArray);
    }

    public static void main(String[] args){

        int size;
        System.out.println("Enter the size of array you wish to have :-");
        Scanner element = new Scanner(System.in);
        size = element.nextInt();
        int array[] = new int[size];

        System.out.println("First Defination of array is : " + Arrays.toString(array));
        System.out.println("Enter the elements of the array");

        for (int i = 0; i < size; i++) {
            array[i] = element.nextInt();
        }

        element.close();
        System.out.println("Defination of array after inserting elements : " + Arrays.toString(array));

        int sortedArray[] = sorting(array);

        System.out.println("New Sorted array is : " + Arrays.toString(sortedArray));

    }

}

File: introductionToAlgorithmCLRS/chp2/Exercise/Ex_2.1/q2RevInsertionSort.py
def insertionSort(array = []):
    sortedArray = array
    for i in range(len(array)):
        key = sortedArray[i]
        j = i - 1
        while (j > -1 and sortedArray[j] < key):
            sortedArray[j + 1] = sortedArray[j]
            j = j - 1
        sortedArray[j + 1] = key
    return sortedArray

print("Enter the size of Array : ", end="")
size = int(input())

print("Enter the elements of the array : ", end="")
array = list(map(lambda x: int(x), input().split()))

sortedArray = insertionSort(array)

print("New Sorted Array is : ", sortedArray)
